src/libstrongswan/plugins/wolfssl/wolfssl_plugin.c

```diff
@@ -47,6 +47,8 @@
 
 #include <wolfssl/ssl.h>
 
+#undef RNG
```

Comment:
There is already such an `#undef` in `wolfssl_common.h`: https://github.com/strongswan/strongswan/blob/84166508f8f0071e7a19f85dadfdd4ef12c5736b/src/libstrongswan/plugins/wolfssl/wolfssl_common.h#L84 So I wondered what exact combination of configure options is necessary to actually trigger a conflict here as we didn't run into it in our CI builds.

As it turns out, our builds enable EdDSA and the headers we include in `wolfssl_common.h` then include `wolfssl/wolfcrypt/random.h`, which defines `RNG`. I guess the OpenWrt build of wolfSSL doesn't enable Ed25519 or Ed448 (or FIPS mode) so this doesn't happen and the definition happens later when `ssl.h` is included.

An alternative would be to define `NO_OLD_RNGNAME` in `wolfssl_common.h` before including any wolfSSL headers, as that seems to prevent the definition of `RNG` [here](https://github.com/wolfSSL/wolfssl/blob/d930825a924efb22e0631dd6a963292040a4d00c/wolfssl/wolfcrypt/random.h#L192-L196). However, I think the `#undef` at the end would still be necessary if FIPS mode is enabled as that has it's own definition of `RNG` [here](https://github.com/wolfSSL/wolfssl/blob/d930825a924efb22e0631dd6a963292040a4d00c/wolfssl/wolfcrypt/settings.h#L326-L341) (our headers don't actually define `RNG`, it's just an implicit macro argument, so the check there doesn't help).

Anyway, since the conflict is only in `wolfssl_plugin.c`, the simplest fix is probably to remove the `#undef` from `wolfssl_common.h` and just do it here. Could you please update the commit accordingly? I'll add some context to the commit message before merging.

Reply:
Done.